src/H5Opline.c

```diff
@@ -110,6 +110,14 @@ H5FL_DEFINE(H5O_pline_t);
  *
  *-------------------------------------------------------------------------
  */
+
+#define VERIFY_LIMIT(p,s,l)                                                  \
+  if (p + s - 1 > l) {                                                       \
+    HGOTO_ERROR(H5E_RESOURCE, H5E_NOSPACE, NULL,                             \
+                "ran off the end of the buffer: current p = %p, p_end = %p", \
+                p + s, l);                                                   \
+  };
+
```

Comment:
I am not a fan of adding a local convenience macro here.

Reply:
I've removed the macro, now, and included explicit tests.